ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh

```diff
@@ -1,21 +1,64 @@
 #!/usr/bin/env bash
 
 set -x
+set +e
+
+BASE_DIR=${HOME:-"/alabama"}
+MUST_GATHER_TIMEOUT=${MUST_GATHER_TIMEOUT:-"5m"}
+
+function create_out_dir {
+  local OUT_DIR=${1}
+  if [ ! -d "./${OUT_DIR}" ]; then
+    mkdir ${OUT_DIR}
+  fi
+}
+
+function gen_pods_debug_cmd {
+  create_out_dir ${1}
+  local cmd=("echo 'Logs for pod: {}'; oc describe pod {} > ${1}/{}-describe.log; oc logs --all-containers=true {} >> ${1}/{}-containers.log")
+  echo "${cmd[@]}"
+}
+
+function gen_api_resources_debug_cmd {
+  create_out_dir ${1}
+  local cmd=("echo 'Info for api-resource: {}'; oc get {} -o yaml > ${1}/{}.log")
+  echo "${cmd[@]}"
+}
 
 # We don't want to use OpenShift-CI build cluster namespace
 unset NAMESPACE
-
 oc project openstack
 
-oc logs ceph
+pushd "${BASE_DIR}" || exit
+mkdir logs
+pushd logs || exit
 
-oc get OpenStackControlPlane openstack -o json
+oc get all > get_all.log
 
-oc get all
+CMD=$(gen_api_resources_debug_cmd "resources")
+oc api-resources | grep 'openstack' | awk '{print $1}' | xargs -n1 -I {} sh -c "${CMD[@]}"
+# RabbitMQCluster
+oc get -o yaml RabbitMQCluster > RabbitMQCluster.log
 
-oc get -o yaml MariaDB,RabbitMQCluster,KeystoneAPI,PlacementAPI,Glance,Cinder,NeutronAPI,Nova
+CMD=$(gen_pods_debug_cmd "controller-manager")
+oc get pods -n openstack --show-labels | grep -i '.*control\-plane\=.*controller\-manager.*' | awk '{print $1}' | \
```

Comment:
Ack, this will affect all steps in the end. But I can already parameterize both here, operators-ns and services-ns. I will provide an update for that.